builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Security Considerations
+description: A description of the main differences that Ethereum developers need to understand in terms of security considerations when developing on Moonbeam.
+---
+
+# Security Considerations
+
+![Moonbeam v Ethereum - Security Banner](/images/builders/get-started/eth-compare/security/security-banner.png)
+
+## Introduction {: #introduction }
+
+When developing smart contracts on Moonbeam, there are some security considerations to be aware of that do not apply when developing on Ethereum. Moonbeam has several [precompiled contracts](/builders/pallets-precompiles/precompiles/){target=_blank}, which are Solidity interfaces that enable developers to access Substrate-based functionality through the Ethereum API, but circumventing the EVM. Although the precompiled contracts are designed to improve the developer experience, there can be some unintended consequences that must be considered.
+
+This guide will outline and provide examples of some security considerations to be cognizant of when developing on Moonbeam.
+
+## Arbitrary Code Execution {: #arbitrary-code-execution }
+
+Arbitrary code execution in Solidity is the ability to execute code and call functions of other contracts using an arbitrary number of arguments of any type.
+
+A smart contract allows arbitrary execution of another contract when it allows a user to invoke the `call()` function and pass in arbitrary call data. The [`call()` function](https://solidity-by-example.org/call/){target=_blank} is made available through the [address data type in Solidity](https://docs.soliditylang.org/en/latest/types.html#address){target=_blank}. When the `call()` function is invoked, the target contract is called using the arbitrary call data.
+
+An example of arbitrary code execution, in it's simplest form, can be seen in the following diagram.
```

Comment:
```suggestion
Arbitrary code execution follows the pattern in the diagram below when `Contract A` allows a user to influence its call to `Contract B`.
```

This might help make it clear what `Contract A` must do "wrong" in order to have potential issues.